### ISSUE TITLE
Ability to embargo an object until a specific date (with auto-release)

### ISSUE THREAD

**Issue by elizoller** [asulibraries/islandora-repo#44]
Question about full text searching and embargoes - do you want embargoed items to be full text searchable (with text snippet previews)?

**Comment by elizoller** [comment]
do you think we could use https://www.drupal.org/project/scheduler? it provides helpful hooks that i think we might be able to advantage of?
see [scheduler.api.php](https://git.drupalcode.org/project/scheduler/blob/8.x-1.x/scheduler.api.php)

**Comment by elizoller** [comment]
Related issues:
https://github.com/Islandora/documentation/issues/1161
https://github.com/Islandora/documentation/issues/928

https://github.com/mjordan/ip_range_access

**Comment by elizoller** [comment]
Add a field on the media for Embargo Date. Add a filter on the EVA view for OSD, PDFjs, etc that is less than today.

**Comment by elizoller** [comment]
The field default value should be none.
The view should have a separate group for if the date is NULL OR less than today

<img width="797" alt="Screen Shot 2020-02-07 at 5 15 15 PM" src="https://user-images.githubusercontent.com/5439169/74074893-6d5a9e00-49cd-11ea-9ef4-0ad0344a143e.png">

**Comment by elizoller** [comment]
Bryan Brown brought up some good points about what if an embargo is added later and someone already had the link to the file. I think we should tie embargoing in more with Drupal's notion of published/unpublished to accommodate that. We may need an additional module like https://www.drupal.org/project/scheduler or  https://www.drupal.org/project/scheduled_publish to facilitate that. I tested scheduled_publish and it works with workflow to be able to add a date/time for content to move from one workflow state to another so I prototyped a concept where there is an Embargo workflow with two states Embargoed or Published and it is attached to all media types with a scheduled publish field on those media types that would allow embargo dates to be set. They would be "released" via cron. This would allow the node to still be accessible but not the media. It may require development of an additional view for display selected data about unpublished/embargoed objects.
